src/plan.rs

```diff
@@ -135,7 +135,7 @@ impl ExecutionPlan {
             let mut image = wm_try!(decode(&file_plan.filename, file_plan.format));
 
             for operation in &file_plan.ops {
-                operation.execute(&mut image)?;
+                operation.execute(&file_plan.filename, &mut image)?;
```

Comment:
We could also dream up a `OperationContext` or something like that as

```rust
struct OperationContext {
  filename,
  image
}
```

and I hope I didn't miss seeing any existing type that contains both the info 👼🏽

Reply:
I think we should add `properties: InputProperties` to the `Image` struct and put things like filename, [`ExtendedColorType`](https://docs.rs/image/latest/image/enum.ExtendedColorType.html) we get from [`original_color_type()`](https://docs.rs/image/latest/image/trait.ImageDecoder.html#method.original_color_type) and so on into it.

Reply:
Cool, yeah, let me explore that.